mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -923,11 +935,24 @@ static LogicalResult printFunctionArgs(CppEmitter &emitter,
                                        Region::BlockArgListType arguments) {
   raw_indented_ostream &os = emitter.ostream();
 
-  return (interleaveCommaWithError(
-      arguments, os, [&](BlockArgument arg) -> LogicalResult {
-        return emitter.emitVariableDeclaration(
-            functionOp->getLoc(), arg.getType(), emitter.getOrCreateName(arg));
-      }));
+  if (auto emitCFunc = dyn_cast<emitc::FuncOp>(functionOp)) {
```

Comment:
```suggestion
  if (auto emitCFunc = dyn_cast<emitc::FuncOp>(functionOp)) {
```

Also works with `func::FuncOp`, factor this